packages/grafana-ui/src/components/Graph/GraphContextMenu.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useContext } from 'react';
+import { ContextMenu, ContextMenuProps } from '../ContextMenu/ContextMenu';
+import { ThemeContext } from '../../themes';
+import { SeriesIcon } from '../Legend/SeriesIcon';
+import { GraphDimensions } from './GraphTooltip/types';
+import {
+  DateTimeInput,
+  FlotDataPoint,
+  getValueFromDimension,
+  getDisplayProcessor,
+  formattedValueToString,
+  Dimensions,
+} from '@grafana/data';
+import { css } from 'emotion';
+
+export type ContextDimensions<T extends Dimensions = any> = { [key in keyof T]: [number, number | undefined] | null };
+
+export type GraphContextMenuProps = ContextMenuProps & {
+  getContextMenuSource: () => FlotDataPoint | null;
+  formatSourceDate: (date: DateTimeInput, format?: string) => string;
+  dimensions?: GraphDimensions;
+  contextDimensions?: ContextDimensions;
+};
+
+export const GraphContextMenu: React.FC<GraphContextMenuProps> = ({
+  getContextMenuSource,
+  formatSourceDate,
+  items,
+  dimensions,
+  contextDimensions,
+  ...otherProps
+}) => {
+  const theme = useContext(ThemeContext);
+  const source = getContextMenuSource();
+
+  //  Do not render items that do not have label specified
+  const itemsToRender = items
+    ? items.map(group => ({
+        ...group,
+        items: group.items.filter(item => item.label),
+      }))
+    : [];
+
+  const renderHeader = () => {
+    if (!source) {
+      return null;
+    }
+
+    // If dimensions supplied, we can calculate and display value
+    let value;
+    if (dimensions && contextDimensions && contextDimensions.yAxis && contextDimensions.yAxis[1]) {
+      const valueFromDimensions = getValueFromDimension(
+        dimensions.yAxis,
+        contextDimensions.yAxis[0],
+        contextDimensions.yAxis[1]
+      );
+      const display = source.series.valueField.display ?? getDisplayProcessor({ field: source.series.valueField });
+      value = display(valueFromDimensions);
+    }
```

Comment:
If dimensions provided, calculate and display value.  Otherwise value will be not calculated and rendered.